### PR TITLE
#558 - Workaround for enum mapping

### PR DESCRIPTION
**Pull request by Thuen**
## Description
- Workaround for DelegationChangeType enum
- Removed CTE from insert query

## Related Issue(s)
- #558 

## Developer/Reviewer Checklist
- [ ] **Your** code builds clean without any errors or warnings
- [ ] No changes to config/appsettings or environment variables created in separate linked PR
- [ ] Manual testing done (required)
- [ ] Relevant integration test added (required for functional changes)
- [ ] Relevant automated test added (required for functional changes)
- [ ] All integration tests run green

## Documentation
- [ ] User documentation is updated with a separate linked PR in [altinn-studio-docs.](https://github.com/Altinn/altinn-studio-docs) (if applicable)

### CODE REVIEW

File: src/Altinn.AccessManagement.Persistence/DelegationMetadataRepo.cs [hunk 1]
```diff
@@ -199,14 +199,28 @@ public async Task<DelegationChange> InsertDelegation(ResourceAttributeMatchType
 
             return await InsertResourceRegistryDelegation(delegationChange, cancellationToken);
         }
-
+        private string TranslateChangeType(DelegationChangeType type)
```
Comment (jonkjetiloye) [inline src/Altinn.AccessManagement.Persistence/DelegationMetadataRepo.cs:202]: Må nesten fikse disse "Closing brace should be followed by blank line" så build and test kjører

**Review by lovoll — APPROVED** [review]
Samme om JK. Fikse klagene på style

File: src/Altinn.AccessManagement.Persistence/DelegationMetadataRepo.cs [hunk 2]
```diff
@@ -255,27 +269,22 @@ private async Task<DelegationChange> InsertResourceRegistryDelegation(Delegation
...
+            @coveredByUserId,
+            @coveredByPartyId,
+            @performedByUserId,
+            @performedByPartyId,
+            @blobStoragePolicyPath,
+            @blobStorageVersionId,
+            @delegatedTime
+            FROM accessmanagement.Resource AS R 
+            WHERE resourceRegistryId = @resourceregistryid
+            RETURNING *
```
Comment (jonkjetiloye) [inline src/Altinn.AccessManagement.Persistence/DelegationMetadataRepo.cs:287]: med Returning * her nå vil man da ikke lenger få tilbake ResourceRegistryId eller ResourceType noe function varianten gjorde

**Comment by andreasisnes** [comment]
Setting it to close after consent from Marius

**Pull request closed without merging**